init: add pdp project examples (#5854)
prisma/prisma-examples: prisma-data-platform/pulse/email-with-resend/README.md

> # Prisma Pulse and Resend Integration Example

Reviewer: a few general thoughts on the README:
- I think we should strive for consistency across READMEs, in this case I would love to see similar structures/wording between the starter examples and this one
- I _think_ this README lacks instructions to run `prisma generate`, right? when I follow the instructions as outlined in the README, I won't be able to add new records to the `User` table because it doesn't exist yet?

This repository contains an example app that uses Prisma Pulse to detect when new users are added to the database and sends them an onboarding email using [Resend](https://resend.com/).

> ---

Reviewer: ```suggestion
```

> ## Prerequisite

Reviewer: ```suggestion
## Prerequisites
```

> 1. A Prisma Pulse-compatible PostgreSQL database.
>     > You can learn how to setup a Pulse-ready database by reading our [docs](https://www.prisma.io/docs/pulse/database-setup).

Reviewer: ```suggestion
1. A Prisma Pulse-compatible PostgreSQL database. If you don't have one yet, you can learn how to set it up in our [docs](https://www.prisma.io/docs/pulse/database-setup).
```

2. Visit [Prisma Data Platform](https://pris.ly/pdp) and enable Pulse in your project environment to get a Pulse API key.

> 3. Create a [Resend account](https://resend.com/) and get the Resend API Key.

Reviewer: I notice several inconsistencies/issues here:
- usage of numbered bullets but is the order of these steps important?
- step 1. mentions "what's needed", step 2. and 3. are "instructions"

I think we should strive for consistency here.

Reviewer: a better version could be smth like:
- A Prisma Pulse-compatible PostgreSQL database. If you don't have one yet, you can learn how to set it up in our [docs](https://www.prisma.io/docs/pulse/database-setup).
- A Pulse API key. If you don't have one yet, you can obtain it in your [Prisma Data Platform](https://pris.ly/pdp) project.
- A Resend API key. If you don't have one yet, you can get it by logging into your [Resend](https://resend.com/) account.

Reviewer: ideally this could be made consistent with the prerequisites in the other Pulse example though: [add-pdp-examples/prisma-data-platform/pulse/starter#prerequisites](https://github.com/prisma/prisma-examples/tree/add-pdp-examples/prisma-data-platform/pulse/starter#prerequisites)

<img width="674" alt="image" src="https://github.com/prisma/prisma-examples/assets/4058327/6276cf68-4de4-469a-8003-0feb42fd11e8">

## Getting Started

To get started, follow these steps:

1. Clone the repository, navigate into it and install dependencies:

    ```bash
    git clone git@github.com:prisma/prisma-examples.git --depth=1
    cd prisma-examples/prisma-data-platform/pulse/email-with-resend
    npm install
    ```

2. Create a `.env` file in the root directory and provide your Database URL, Prisma Pulse and Resend API key:

   ```bash
   DATABASE_URL=""
   PULSE_API_KEY=""
   RESEND_API_KEY=""
   ```

## Run the app

1. Start the script using:

   ```bash
   npm run dev
   ```

> 2. Add a new user record with a valid email to the `User` table in the database using Prisma Studio or any other database tool:

Reviewer: ```suggestion
2. Add a new record with a valid email to the `User` table in the database using Prisma Studio (or any other database tool):
```

   ```bash

>    npm prisma studio

Reviewer: ```suggestion
   npx prisma studio
```

   ```

3. Add a new user with a `name` and a *valid* `email`.

> 4. You will receive a new email in the insterted email inbox. For example: When a new user with an email of `datta@prisma.io` and a name of `Ankur` is created, the email inbox for `datta@prisma.io` should have a new email:

Reviewer: ```suggestion
4. You will receive a new email in the inbox of the specified email. For example: When a new user with an email of `datta@prisma.io` and a name of `Ankur` is created, the email inbox for `datta@prisma.io` should have received the following email (the copy for this email is hardcoded in [`./script.ts`](./script.ts#L33)):
```

   ![image.png](./images/email.png)

## Next steps

> - Checkout the Prisma Pulse [docs](https://www.prisma.io/docs/pulse).

Reviewer: ```suggestion
- Checkout the [Prisma Pulse docs](https://www.prisma.io/docs/pulse).
```
